File: backend/app/models/tree.py
from pydantic import BaseModel
from typing import Optional, Dict, List


class Option(BaseModel):
    """選択肢"""
    id: str
    label: str
    next_node_ids: Optional[List[str]] = None  # 複数の次ノードID


class Hint(BaseModel):
    """ヒント"""
    text: str
    type: str  # 'command' or 'text'


class Node(BaseModel):
    """ツリーのノード"""
    id: str
    question: str
    hint: Optional[str] = None  # 後方互換性のため残す
    hint_type: Optional[str] = None  # 後方互換性のため残す
    hints: Optional[List[Hint]] = None  # 新しいヒント形式
    type: str  # 'select', 'multiselect', 'text'
    options: List[Option]
    description: Optional[str] = None


class Tree(BaseModel):
    """ツリー全体"""
    id: str
    title: str
    description: str
    root_node_id: str
    nodes: Dict[str, Node]

File: frontend/src/types/tree.ts
// バックエンドのPydanticモデルに対応する型定義

export interface Option {
  id: string;
  label: string;
  next_node_ids?: string[];  // 複数の次ノードID
}

export interface Hint {
  text: string;
  type: 'command' | 'text';
}

export interface Node {
  id: string;
  question: string;
  hint?: string;  // 後方互換性のため残す
  hint_type?: 'command' | 'text';  // 後方互換性のため残す
  hints?: Hint[];  // 新しいヒント形式
  type: string;  // 'select', 'multiselect', 'text'
  options: Option[];
  description?: string;
}

export interface Tree {
  id: string;
  title: string;
  description: string;
  root_node_id: string;
  nodes: Record<string, Node>;
}
